src/components/navigation/SlideMenu.tsx

```diff
@@ -1,156 +1,69 @@
-  const classes = useStyles();
-  const history = useHistory();
-  const hasSubList = !!props.subItemList?.length;
-  const isSelected: boolean = history.location.pathname === props.path;
-
-  if (typeof props.isVisible === 'function' && !props.isVisible()) {
-    return null;
-  }
-
-  const handleClick = (): void => {
-    if (hasSubList) {
-      if (open) {
-        menuState.expandedNavItems.delete(props.id);
-        setOpen(false);
-      } else {
-        menuState.expandedNavItems.add(props.id);
-        setOpen(true);
-      }
-    } else {
-      if (typeof props.path === 'string') {
-        history.push(props.path);
-        props.menuClosingAction();
-      }
-    }
-  };
-
-  const expandButton = open ? <ExpandLess /> : <ExpandMore />;
-
-  const subList: React.ReactNode = hasSubList ? (
-    <Collapse in={open} timeout="auto">
-      <List className={classes.nestedListItem} disablePadding>
-        {props.subItemList
-          ?.map((item: NavigationItem) => (
-            <NavListItem key={item.id} {...item} menuClosingAction={props.menuClosingAction} />
-          ))
-          .filter(Boolean)}
-      </List>
-    </Collapse>
-  ) : null;
-
-  return (
-    <>
-      <ListItem onClick={handleClick} selected={isSelected} button>
-        {props.icon ? <ListItemIcon>{props.icon}</ListItemIcon> : null}
-        <ListItemText primary={props.text} />
-        {hasSubList ? expandButton : null}
-      </ListItem>
-      {subList}
-    </>
-  );
-};
-
-interface UtilityListItemProps extends UtilityItem {
-  menuClosingAction: () => void;
-}
-const UtilityListItem: React.FC<UtilityListItemProps> = props => {
-  const handleClick = (): void => {
-    if (props.clickBehavior === 'disclose') {
-      // open modal
-      props.menuClosingAction();
-    } else if (props.clickBehavior === 'link') {
-      if (props.url) {
-        props.menuClosingAction();
-        window.open(props.url, '_blank');
-      }
-    }
-  };
-
-  return (
-    <ListItem onClick={handleClick} button>
-      {props.icon ? <ListItemIcon>{props.icon}</ListItemIcon> : null}
-      <ListItemText primary={props.text} />
-    </ListItem>
-  );
-};
+import useStyles from './SlideMenu.jss';
 
+/**
+ * Slide menu disclosed in the drawer. Contains navigation and utility lists.
+ */
 interface SlideMenuProps {
+  /**
+   * Determines whether the menu drawer is open
+   */
   open: boolean;
+  /**
+   * Callback for when an opening action is taken. function().
+   */
+  onRequestOpen: () => void;
+  /**
+   * Callback for when a closing action is taken. function().
+   */
   onRequestClose: () => void;
 }
 
 const SlideMenu: React.FC<SlideMenuProps> = props => {
-  const { open, onRequestClose } = props;
+  const { open, onRequestOpen, onRequestClose } = props;
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('xs'));
   const classes = useStyles();
+  const drawerClass: string = clsx(classes.drawer, {
```

Comment:
the compiler config is smart enough to infer this type, there is never really a need to specify primitive types
```suggestion
  const drawerClass = clsx(classes.drawer, {
```

Reply:
Removed https://github.com/resident51/resident51/pull/3/commits/ad9cf714da9c80a0a85a88d5891d7dee55f45ffa